packages/block-editor/src/components/child-layout-control/index.js

```diff
@@ -51,49 +64,83 @@ export default function ChildLayoutControl( {
 
 	return (
 		<>
-			<ToggleGroupControl
-				__nextHasNoMarginBottom
-				size={ '__unstable-large' }
-				label={ childLayoutOrientation( parentLayout ) }
-				value={ selfStretch || 'fit' }
-				help={ helpText( selfStretch, parentLayout ) }
-				onChange={ ( value ) => {
-					const newFlexSize = value !== 'fixed' ? null : flexSize;
-					onChange( {
-						...childLayout,
-						selfStretch: value,
-						flexSize: newFlexSize,
-					} );
-				} }
-				isBlock={ true }
-			>
-				<ToggleGroupControlOption
-					key={ 'fit' }
-					value={ 'fit' }
-					label={ __( 'Fit' ) }
-				/>
-				<ToggleGroupControlOption
-					key={ 'fill' }
-					value={ 'fill' }
-					label={ __( 'Fill' ) }
-				/>
-				<ToggleGroupControlOption
-					key={ 'fixed' }
-					value={ 'fixed' }
-					label={ __( 'Fixed' ) }
-				/>
-			</ToggleGroupControl>
-			{ selfStretch === 'fixed' && (
-				<UnitControl
-					size={ '__unstable-large' }
-					onChange={ ( value ) => {
-						onChange( {
-							...childLayout,
-							flexSize: value,
-						} );
-					} }
-					value={ flexSize }
-				/>
+			{ parentLayoutType === 'flex' && (
+				<>
+					<ToggleGroupControl
+						__nextHasNoMarginBottom
+						size={ '__unstable-large' }
+						label={ childLayoutOrientation( parentLayout ) }
+						value={ selfStretch || 'fit' }
+						help={ helpText( selfStretch, parentLayout ) }
+						onChange={ ( value ) => {
+							const newFlexSize =
+								value !== 'fixed' ? null : flexSize;
+							onChange( {
+								...childLayout,
+								selfStretch: value,
+								flexSize: newFlexSize,
+							} );
+						} }
+						isBlock={ true }
+					>
+						<ToggleGroupControlOption
+							key={ 'fit' }
+							value={ 'fit' }
+							label={ __( 'Fit' ) }
+						/>
+						<ToggleGroupControlOption
+							key={ 'fill' }
+							value={ 'fill' }
+							label={ __( 'Fill' ) }
+						/>
+						<ToggleGroupControlOption
+							key={ 'fixed' }
+							value={ 'fixed' }
+							label={ __( 'Fixed' ) }
+						/>
+					</ToggleGroupControl>
+					{ selfStretch === 'fixed' && (
+						<UnitControl
+							size={ '__unstable-large' }
+							onChange={ ( value ) => {
+								onChange( {
+									...childLayout,
+									flexSize: value,
+								} );
+							} }
+							value={ flexSize }
+						/>
+					) }
+				</>
+			) }
+			{ parentLayoutType === 'grid' && (
+				<HStack>
+					<InputControl
+						size={ '__unstable-large' }
+						label={ __( 'Column Span' ) }
+						type="number"
+						onChange={ ( value ) => {
+							onChange( {
+								...childLayout,
+								columnSpan: value,
+								parentColumnWidth,
```

Comment:
`parentColumnWidth` is exactly the parent's `minimumColumnWidth`. It's being added as an attribute here so we can access it when `render_layout_support_flag` runs, because at that point we only know about the block being processed, and have no access to its ancestor tree.

Reply:
If it's being cached here will that be a problem if someone goes to update the parent column width after they've set the row/column span, or if they drag and drop between two grids?

Reply:
I _think_ it's being cached after drag. I'm testing some nested layouts and have dragged a child to another grid parent and it still has the value of the former parent's `minimumColumnWidth` in `parentColumnWidth`

```json
"layout":{"columnSpan":"3","parentColumnWidth":"300px"}}
```

Is that what you mean?

Reply:
Yep, that's what I was getting at. The `parentColumnWidth` should be the current parent's column width, but the approach right now means that it only gets set when the column and row span values are updated 🤔

Reply:
Oh sorry I didn't see this one. It seems pretty unlikely someone will just drag and drop a block with a column span from one grid to another differently sized grid and not make any further adjustments (and if both grids are the same size it won't be an issue) but I'm open to suggestions if anyone can find a better way to do it!

Even if someone does drag and drop a multi-column block between differently sized grids the very worst that can happen is the container query gets a bit wonky and doesn't break at exactly the point it should 🤷

Reply:
> so we can access it when `render_layout_support_flag` runs, because at that point we only know about the block being processed, and have no access to its ancestor tree.

Oh really? Bugger. That seems like a shortcoming of the `render_block` filter that's maybe worth fixing. We maybe could work around it by using `render_block_data` which is passed `$parent_block`.

https://github.com/WordPress/wordpress-develop/blob/8ec4d9dfc7e2043485d83d4af9e505ec1cc21470/src/wp-includes/class-wp-block.php#L442


```php
add_filter( 'render_block_data', function( $parsed_block, $source_block, $parent_block ) {
    $parsed_block['parent_block'] = $parent_block;
}, 10, 3 );

add_filter( 'render_block', function( $block_content, $block ) {
    var_dump($block['parent_block']);
}, 10, 2 );
```

Reply:
To be clear, is that something you think should be addressed _now_, as part of this PR?

Reply:
> To be clear, is that something you think should be addressed now, as part of this PR?

IMO, I don't think it'd be a blocker to landing, but would be good to look at in a follow-up before Grid is taken out of experimental.

Reply:
Ok I might do this as a follow-up because if we want to remove the `parentColumnWidth` attrib altogether it'll need some work in both editor and front-end logic.